hpx/util/tuple.hpp

```diff
@@ -656,6 +721,24 @@ namespace hpx { namespace util
         }
     };
 
+    template <std::size_t I, typename ...Ts>
+    struct tuple_element<I, std::tuple<Ts...>>
+    {
+        using type = typename std::tuple_element<I, std::tuple<Ts...>>::type;
+
+        static HPX_CONSTEXPR HPX_HOST_DEVICE HPX_FORCEINLINE type&
+        get(std::tuple<Ts...>& tuple) noexcept
+        {
+            return std::get<I>(tuple);
+        }
+
+        static HPX_CONSTEXPR HPX_HOST_DEVICE HPX_FORCEINLINE type const&
+        get(std::tuple<Ts...> const& tuple) noexcept
+        {
+            return std::get<I>(tuple);
+        }
+    };
```

Comment:
Overloads for rvalue refs?

Reply:
Not here, the two rvalue ref overloads are synthesized by `util::get`.